util/stmtsummary/statement_summary.go

```diff
@@ -341,8 +342,13 @@ func (ssMap *stmtSummaryByDigestMap) GetMoreThanOnceSelect() ([]string, []string
 				if ssbd.history.Len() > 0 {
 					ssElement := ssbd.history.Back().Value.(*stmtSummaryByDigestElement)
 					ssElement.Lock()
+					sampleUser := ""
+					for key := range ssElement.authUsers {
+						sampleUser = key
+						break
+					}
 					// Empty sample users means that it is an internal queries.
-					if ssElement.sampleUser != "" && (ssbd.history.Len() > 1 || ssElement.execCount > 1) {
+					if sampleUser != "" && (ssbd.history.Len() > 1 || ssElement.execCount > 1) {
```

Comment:
Since "" is also put into `authUsers`, `sampleUser` can still be "" even if there's other users in `authUsers`.
You shouldn't put "" into `authUsers`, and this block can be simply `if len(ssElement.authUsers) > 0 && ...`.